uaclient/files.py

```diff
@@ -55,6 +66,52 @@ def delete(self):
         system.remove_file(self.path)
 
 
+class DataObjectFileFormat(Enum):
+    JSON = object()
+    YAML = object()
+
+
+DOFType = TypeVar("DOFType", bound=data_types.DataObject)
+
+
+class DataObjectFile(Generic[DOFType]):
+    def __init__(
+        self,
+        data_object_cls: Type[DOFType],
+        ua_file: UAFile,
+        file_format: DataObjectFileFormat = DataObjectFileFormat.JSON,
+    ):
+        self.data_object_cls = data_object_cls
+        self.ua_file = ua_file
+        self.file_format = file_format
+
+    def read(self) -> Optional[DOFType]:
+        raw_data = self.ua_file.read()
+        if raw_data is None:
+            return None
+
+        parsed_data = None
+        if self.file_format == DataObjectFileFormat.JSON:
+            parsed_data = json.loads(raw_data)
+        elif self.file_format == DataObjectFileFormat.YAML:
+            parsed_data = yaml.safe_load(raw_data)
```

Comment:
Should we perform any exception handling for invalid json or yaml content ?

Reply:
Yeah we probably should - good catch! We should probably just catch either json or yaml decode error and re-raise it as a "InvalidFormatError" so that the caller can just catch that without know what the underlying format is.